parser.go

```diff
@@ -217,103 +268,101 @@ func (p *Parser[V]) Replace(v V) V {
 	return oldVal
 }
 
-// RotateLeft restructures the tree by moving the current node to the
-// position of its parent node. The original parent node becomes a
-// child of the current node. All other child nodes of the original
-// parent remain unchanged, except for the current node which becomes
-// the new parent. If RotateLeft is called on the root node then
-// ErrMissingRequiredNode is returned.
+// RotateLeft performs a left rotation in the case of a binary tree at the
+// current tree location and returns the new root of the rotated sub-tree.
+// If the current node has no right child, this method is a no-op.
 //
-// Note that the default empty root node may be rotated if RotateLeft is called
-// on a child node of the root node..
-func (p *Parser[V]) RotateLeft() (*Node[V], error) {
-	// n = node , op = original parent , gp = grand parent
-	n := p.node
-
-	op := n.Parent
-	if op == nil {
-		return nil, ErrMissingRequiredNode
-	}
-
-	gp := op.Parent
-
-	// Remove n from op's Children
-	opChildren := op.Children[:0]
-	for _, x := range op.Children {
-		if x != n {
-			opChildren = append(opChildren, x)
-		}
+// See: https://en.wikipedia.org/wiki/Tree_rotation
+func (p *Parser[V]) RotateLeft() *Node[V] {
+	// The tree is rotated as follows. The nodes A, B, C are root nodes of
+	// potential sub-trees.
+	/*
+	 *      P                       Q
+	 *  /       \               /		\
```

Comment:
In the github diff view it looks like there are a couple tabs that should be spaces in between the branches under the `Q`.

Reply:
Yeah, I blame gofmt. Will clean up.